Source/WebCore/rendering/TextPaintStyle.h

```diff
@@ -48,6 +48,7 @@ struct TextPaintStyle {
     Color strokeColor;
     Color emphasisMarkColor;
     float strokeWidth { 0 };
+    bool hasExplicitlySetFillColor { false };
```

Comment:
I would rename this to:
```suggestion
    bool hasExplicitlySetColor { false };
```

because fillColor refers to colors set by `-webkit-text-fill-color` as well (whereas this flag doesn't, it just refers to colors explicitly set by the `color` property).

Reply:
Yeah, it's a bit confusing but the text color corresponds to the `fillColor` in the `TextPaintStyle` struct, which is why I left it as `fillColor` instead of just `color`.

Reply:
It corresponds to `fillColor` in the `TextPaintStyle`, because I think that color takes in account both `-webkit-text-fill-color` and `color`, while the explicitlySet flag only covers `color`.